pkg/deploy/controller/deploymentconfig/controller_test.go

```diff
@@ -245,3 +246,148 @@ func TestHandle_fatalError(t *testing.T) {
 		t.Fatalf("expected a fatal error, got: %v", err)
 	}
 }
+
+// TestHandle_previousCompletedDeployment ensures that an attempt to create a
+// new deployment for a config for which all the previous deployments are
+// in a terminal state (complete, failed) is successful
+func TestHandle_previousCompletedDeployment(t *testing.T) {
```

Comment:
I think you can combine this test with `TestHandle_previousPendingDeployment` by adding an `error` field to the `scenario`; then the scenarios which have terminal deployments would have a `nil` error expectation, and the non-terminal scenarios could have a `transientError` expectation.